### PR TITLE
chore: make general improvements

### PR DESCRIPTION
**Pull request by shmsr**
Signed-off-by: subham sarkar <sarkar.subhams2@gmail.com>

### CODE REVIEW

**Review by mr-karan — CHANGES_REQUESTED** [review]

**Comment by mr-karan** [comment]
Thanks, @shmsr. Mostly looks good, just one minor change.

File: log_test.go [hunk 1]
```diff
@@ -111,7 +111,7 @@ func TestLogFormat(t *testing.T) {
 	buf.Reset()
 
 	// Fatal log
-	var hadExit = false
+	hadExit := false
```
[the remark this answers is not in the document]
Comment (shmsr) [reply log_test.go:114]: Ideally, this should've been:
```go
var hadExit bool // zero value is `false` by default
```

I've always followed this convention after reading: [2 cents by Dave Cheney](https://dave.cheney.net/practical-go/presentations/qcon-china.html#_use_a_consistent_declaration_style). If you check golang/go you would find they mostly take care of it (obviously there are exceptions as these are nitpicks).

So, as in the code both declaration and initialisation were happening, that's why I used `:=`.

Comment (shmsr) [inline log_test.go:114]:
```suggestion
	var hadExit bool
```

Comment (shmsr) [reply log_test.go:114]: Better?

Comment (mr-karan) [reply log_test.go:114]: Yes, this seems bit better than the earlier one. :)

**Pull request merged**